security/password.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package security
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/util"
+	"golang.org/x/crypto/bcrypt"
```

Comment:
Ah, nevermind, that's just what the golang implementation makes it sound like.
Cursory (very cursory) search online shows that it's iffy which one is best, as long as you keep the cost parameters updated. I'm happy searching for better info on that if you think it's worth it.

As for password format, I intend to have everything within security worry about it, everything else should just see a byte slice. If we need to support more than one eventually, we can simply add a "hash type" field to the UserConfig proto.